aea/configurations/loader.py

```diff
@@ -245,20 +281,33 @@ def _load_agent_config(self, file_pointer: TextIO) -> AgentConfig:
             key_order
         )
 
+        component_configurations = self._get_component_configurations(
+            configuration_file_jsons
+        )
+        agent_configuration_obj.component_configurations = component_configurations
+        return agent_configuration_obj
+
+    def _get_component_configurations(
+        self, configuration_file_jsons
+    ) -> Dict[ComponentId, Dict]:
+        """
+        Get the component configurations from the tail pages of the aea-config.yaml file.
+
+        :param configuration_file_jsons: the JSON objects of the custom configurations of a aea-config.yaml file.
+        :return: a dictionary whose keys are component ids and values are the configurations.
+        """
         component_configurations: Dict[ComponentId, Dict] = {}
         # load the other components.
         for i, component_configuration_json in enumerate(configuration_file_jsons[1:]):
```

Comment:
1 should be declared as constant, like `STARTING_INDEX_CUSTOM_CONFIGS` or similar